### PR TITLE
feat(devices): Optical sensor

### PR DESCRIPTION
**Pull request by Tropix126**
👁️

### CODE REVIEW

File: pros-sys/src/optical.rs [hunk 1]
```diff
@@ -0,0 +1,248 @@
+use core::ffi::*;
+use crate::PROS_ERR;
+
+#[doc(hidden)]
+pub const OPT_GESTURE_ERR: i8 = i8::MAX;
```
Comment (Gavin-Niederman) [inline pros-sys/src/optical.rs]: Consider removing this doc hidden.

File: pros/src/sensors/optical.rs [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+pub const MAX_LED_PWM: u8 = 100;
+
+#[derive(Debug)]
+pub struct OpticalSensor {
+    port: u8,
+    gesture_detection_enabled: bool,
+}
+
+impl OpticalSensor {
+    pub fn new(port: u8, gesture_detection_enabled: bool) -> Result<Self, OpticalError> {
```
Comment (Gavin-Niederman) [inline pros/src/sensors/optical.rs:19]: You should add some docs here. You can talk about how leaving gesture detection on when you don't need it is bad for battery life.

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
Just 2 docs related issues.

File: pros/src/sensors/optical.rs (hunk 2 above)
Comment (Tropix126) [reply pros/src/sensors/optical.rs:19]: It's currently unclear if that's true (just a theory that I had), so it'd probably be a good idea to wait until I have a concrete answer from jpearman before I add that detail. I'll add a basic description for now though.

Comment (Gavin-Niederman) [reply pros/src/sensors/optical.rs:19]: Sounds good.

**Review by Gavin-Niederman — APPROVED** [review]
LGTM (Large Goons That Morph)

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
All that needs to be done is for the TODO.md to be updated. Also formatting, but that's less important.

**Pull request merged**